koku/api/provider/view.py

```diff
@@ -141,11 +127,17 @@ def get_queryset(self):
     @never_cache
     def create(self, request, *args, **kwargs):
         """Create a Provider."""
+        provider_type = request.data.get('type', '')
+        if provider_type and Provider.PROVIDER_CASE_MAPPING.get(provider_type.lower()):
+            request.data['type'] = request.data.get('type', '').lower()
         return super().create(request=request, args=args, kwargs=kwargs)
 
     @never_cache
     def update(self, request, *args, **kwargs):
         """Update a Provider."""
+        provider_type = request.data.get('type', '')
```

Comment:
Why set provider type to `''`? Shouldn't we throw a validation error instead?

Reply:
This was a mistake! I think I had it set to '' before I checked for the source type and forgot to remove it. Good catch! - Fixed 👍